storagemarket/deal_execution.go

```diff
@@ -143,35 +143,11 @@ func (p *Provider) execDealUptoAddPiece(ctx context.Context, deal *types.Provide
 				return derr
 			}
 
-			if err := p.transferAndVerify(dh.transferCtx, pub, deal); err != nil {
+			if err := p.transferAndVerify(dh, pub, deal); err != nil {
 				// The transfer has failed. If the user tries to cancel the
 				// transfer after this point it's a no-op.
 				dh.setCancelTransferResponse(nil)
 
-				// Pass through fatal errors
-				if err.retry == smtypes.DealRetryFatal {
-					return err
-				}
-
-				// If the transfer failed because the user cancelled the
-				// transfer, it's non-recoverable
-				if dh.TransferCancelledByUser() {
-					return &dealMakingError{
-						retry: types.DealRetryFatal,
-						error: fmt.Errorf("data transfer cancelled after %d bytes: %w", deal.NBytesReceived, err),
-					}
-				}
-
-				// If the transfer failed because boost was shut down, it's
```

Comment:
@dirkmc shouldn't we leave this block around, i.e. if the context is canceled, do not mark the deal as fatally failed?